### ISSUE TITLE
Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: HDF5Data

### ISSUE THREAD

**Issue by ygjwd12345** [BVLC/caffe#6658]
### Issue summary
[----------] 12 tests from NesterovSolverTest/0, where TypeParam = caffe::CPUDevice<float>
[ RUN      ] NesterovSolverTest/0.TestNesterovLeastSquaresUpdateWithWeightDecayMultiIter
F0105 15:05:20.597390 38280 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: HDF5Data (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Clip, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Power, Python, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, Swish, TanH, Threshold, Tile, WindowData)
*** Check failure stack trace: ***
    @     0x7fd9223115cd  google::LogMessage::Fail()
    @     0x7fd922313433  google::LogMessage::SendToLog()
    @     0x7fd92231115b  google::LogMessage::Flush()
    @     0x7fd922313e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fd91c4794ac  caffe::Net<>::Init()
    @     0x7fd91c47ac0e  caffe::Net<>::Net()
    @     0x7fd91c45423a  caffe::Solver<>::InitTrainNet()
    @     0x7fd91c455705  caffe::Solver<>::Init()
    @     0x7fd91c455a1f  caffe::Solver<>::Solver()
    @           0x7d56a2  caffe::NesterovSolverTest<>::InitSolver()
    @           0x7cf0ce  caffe::GradientBasedSolverTest<>::TestLeastSquaresUpdate()
    @           0x7d06dc  caffe::NesterovSolverTest_TestNesterovLeastSquaresUpdateWithWeightDecayMultiIter_Test<>::TestBody()
    @           0x9470d3  testing::internal::HandleExceptionsInMethodIfSupported<>()
    @           0x9406ea  testing::Test::Run()
    @           0x940838  testing::TestInfo::Run()
    @           0x940915  testing::TestCase::Run()
    @           0x941bef  testing::internal::UnitTestImpl::RunAllTests()
    @           0x941f13  testing::UnitTest::Run()
    @           0x46f1cd  main
    @     0x7fd91b689830  __libc_start_main
    @           0x477089  _start
    @              (nil)  (unknown)
Makefile:532: recipe for target 'runtest' failed
make: *** [runtest] 已放弃 (core dumped)

### System configuration

* Operating system: Ubuntu 16.04
* CUDA version (if applicable): 9
* CUDNN version (if applicable):7 
* BLAS: 
* Python version (if using pycaffe): 2.7
* MATLAB version (if using matcaffe):

**Comment by wavelet2008** [comment]
fuck
